cpp/tests/io/parquet_test.cpp

```diff
@@ -5210,6 +5210,39 @@ TEST_F(ParquetWriterTest, DictionaryAlwaysTest)
   EXPECT_TRUE(used_dict(1));
 }
 
+TEST_F(ParquetWriterTest, DictionaryPageSizeEst)
+{
+  // one page
+  constexpr unsigned int nrows = 20'000U;
+
+  // this test is creating a pattern of repeating then non-repeating values to trigger
+  // a "worst-case" for page size estimation in the presence of a dictionary. have confirmed
+  // that this fails for values over 16 in the final term of `max_RLE_page_size()`.
+  auto elements0  = cudf::detail::make_counting_transform_iterator(0, [](auto i) {
+    if ((i / 5) % 2 == 1) {
```

Comment:
why is the pattern changed every five elements?

Reply:
It worked :rofl: This was totally empirical...I changed the divisor until I started getting overwrites in the page encoder. Printing out the run lengths, it seemed to give literal runs of 8 followed by repeated runs of 2.  Pretty close to ideal for the purposes of the test.

Reply:
> it seemed to give literal runs of 8 followed by repeated runs of 2

Can you please add this info to the comment above?

Reply:
Thinking seriously about it, the pattern is `CCCCCRRRRRCCCCCRRRRR...`, where C changes, R is repeating.  The encoder will take the first `CCCCCRRR` and emit a literal run of 8, followed by a repeated run of `RR`.  Then this repeats with another `CCCCCRRR`, `RR`.  I'd like to say this was rationally designed, but I got lucky :smile: